midend/copyStructures.cpp

```diff
@@ -49,7 +49,11 @@ const IR::Node* RemoveAliases::postorder(IR::P4Parser* parser) {
 
 const IR::Node* RemoveAliases::postorder(IR::P4Control* control) {
     if (!declarations.empty()) {
-        control->controlLocals.append(declarations);
+        // prepend declarations: they must come before any actions
+        // that may have been modified
+        auto vec = new IR::IndexedVector<IR::Declaration>(declarations);
+        vec->append(control->controlLocals);
+        control->controlLocals = *vec;
```

Comment:
Should be able to use

    control->controlLocals.insert(control->controlLocals.begin(), declarations.begin(), declarations.end())

to do the insertion directly at the beginning, or is that not as clear?  Maybe add a `prepend` method to `IndexedVector`?

Reply:
I have added a prepend method to our Vector classes.